include/arc_utilities/thin_plate_spline.hpp

```diff
@@ -0,0 +1,167 @@
+#ifndef THIN_PLATE_SPLINE_HPP
+#define THIN_PLATE_SPLINE_HPP
+
+#include <Eigen/Dense>
+#include <Eigen/QR>
+#include "arc_utilities/arc_exceptions.hpp"
+#include "arc_utilities/eigen_helpers.hpp"
+
+namespace arc_utilities
+{
+    // Note that DIMENSIONS must be a postive integer, not something like `Eigen::Dynamic`
+    template<int DIMENSIONS = 3>
+    class ThinPlateSpline
+    {
+    public:
+        typedef Eigen::Matrix<double, DIMENSIONS,     Eigen::Dynamic> PointSet;
+        typedef Eigen::Matrix<double, DIMENSIONS + 1, Eigen::Dynamic> HomogeneousPointSet;
+
+        EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+        ThinPlateSpline()
+            : solved_(false)
+        {}
+
+        /**
+         * @brief ThinPlateSpline::ThinPlateSpline An arbitrary dimensional thin
+         *  plate spline for interpolation between points.
+         * @param template_points A (DIMINESION x NUMPOINTS) matrix which define
+         *  the starting points that control the warping. I.e. the template is
+         *  warped to the target
+         * @param target_points A (DIMINESION x NUMPOINTS) matrix which define
+         *  the target points that control the warping. I.e. the template is
+         *  warped to the target
+         */
+        template <typename DerivedA, typename DerivedB>
+        ThinPlateSpline(
+                const Eigen::MatrixBase<DerivedA>& template_points,
+                const Eigen::MatrixBase<DerivedB>& target_points)
+            : solved_(false)
+        {
+            setTemplatePoints(template_points);
+            setTargetPoints(target_points);
+            solveForCoeffs();
+        }
+
+        /**
+         * @note Checking for the vailidty of the input data is put off until
+         *  the warping function is solved
+         */
+        template <typename Derived>
+        void setTemplatePoints(
+                const Eigen::MatrixBase<Derived>& template_points)
+        {
+            using namespace Eigen;
+            static_assert(MatrixBase<Derived>::RowsAtCompileTime == DIMENSIONS ||
+                          MatrixBase<Derived>::RowsAtCompileTime == Dynamic,
+                          "INVALID DATA DIMENSIONS FOR template_points");
+
+            homogeneous_template_points_.resize(DIMENSIONS + 1, template_points.cols());
+            homogeneous_template_points_ << template_points, RowVectorXd::Ones(template_points.cols());
```

Comment:
This statement is confusing and should either have parenthesis or split up into multiple lines (comma operator's not commonly seen and can mess up order of operations). Why can you not just use `RowVectorXd::Ones(template_points.cols())` directly?